go/libraries/doltcore/diff/async_differ.go

```diff
@@ -214,13 +215,30 @@ func (kd *keylessDiffer) GetDiffs(numDiffs int, timeout time.Duration) (diffs []
 				return diffs[:idx], more, nil
 			}
 
-			kd.df, kd.copiesLeft, err = convertDiff(d)
+			df, copiesLeft, err := convertDiff(d)
 			if err != nil {
 				return nil, false, err
 			}
+			if pred(&df) {
+				kd.df = df
+			}
```

Comment:
If `&df` doesn't satisfy the predicate, we need to keep looking for a difference that does

```suggestion
			ok := false
			for !ok {
				kd.df, kd.copiesLeft, err = convertDiff(d)
				if err != nil {
					return nil, false, err
				}
				ok = pred(&df)
			}
```
(couldn't highlight 225 because Github sux)